### PR TITLE
refactor: Reuse scheduling components

### PR DESCRIPTION
**Pull request by Shehryar21**
## Notion ticket link
Closes 
[Make Scheduling Flow Resuable](https://www.notion.so/uwblueprintexecs/Update-Scheduling-Flow-to-be-reusable-783e7486db734c898b257dea9345aa08)
[Confirm Details Screen](https://www.notion.so/uwblueprintexecs/Scheduling-Flow-Confirm-Details-641d2b0038d14f58ac32c132c9e11f9f)

## Implementation description
* Made scheduling components to be reusable so that editing donation would use the same components
Styling still needs to be worked on

## Steps to test
1. Check normal creating a donation flow. You can use the preview url and use the seeded user to login: email: test@test.org and pw: password123
2. Edit a scheduled donation which should take to the confirm donation details screen in the scheduling flow
3. Edit buttons on donation details screen should take the user to the respective screen that they are trying to edit

https://user-images.githubusercontent.com/37384882/142585100-ea436908-98d3-4bbc-979b-0827b2c0fa4c.mov

## What should reviewers focus on?
* 

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/pages/Scheduling/index.tsx [hunk 1]
```diff
@@ -30,23 +31,36 @@ interface UseStepType {
...
-    pickupLocation: "",
-    startTime: new Date().toDateString(),
-    endTime: new Date().toDateString(),
-    status: "",
-    volunteerNeeded: false,
-    frequency: "",
-    notes: "",
-  });
+interface SchedulingProps {
+  schedulingData?: SchedulingFormProps;
```
Comment (hanlinc27) [inline frontend/src/components/pages/Scheduling/index.tsx:35]:
```suggestion
  schedulingData: SchedulingFormProps;
```
this should be mandatory, either the default data or the one being currently edited, right?

File: frontend/src/components/pages/Scheduling/index.tsx [hunk 2]
```diff
@@ -72,6 +87,7 @@ const Scheduling = () => {
           formValues={schedulingFormValues}
           setForm={setSchedulingForm}
           navigation={navigation}
+          isBeingEdited={isBeingEdited}
```
[the remark this answers is not in the document]
Comment (Shehryar21) [reply frontend/src/components/pages/Scheduling/index.tsx:90]: that makes it true even for the default scenario

[the remark this answers is not in the document]
Comment (hanlinc27) [reply frontend/src/components/pages/Scheduling/index.tsx:90]: for the default scenario you can pass in `false` but for the other instances we can pass in `isBeingEdited` directly

[the remark this answers is not in the document]
Comment (Shehryar21) [reply frontend/src/components/pages/Scheduling/index.tsx:90]: not exactly sure what you mean but `isBeingEdited={isBeingEdited}` passes in `false` for default scenario and `true` when being edited which is what we want.

File: frontend/src/components/pages/Scheduling/ConfirmDetails.tsx [hunk 3]
```diff
@@ -1,36 +1,172 @@
...
+import AuthContext from "../../../contexts/AuthContext";
+import { DonorResponse } from "../../../types/DonorTypes";
 import SchedulingProgressBar from "../../common/SchedulingProgressBar";
 import { SchedulingStepProps } from "./types";
 
 const ConfirmDetails = ({
   formValues,
   setForm,
   navigation,
+  isBeingEdited,
```
Comment (hanlinc27) [inline frontend/src/components/pages/Scheduling/ConfirmDetails.tsx:27]:
```suggestion
  isBeingEdited=false,
```

File: frontend/src/components/pages/Scheduling/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (hanlinc27) [reply frontend/src/components/pages/Scheduling/index.tsx:90]: for default scenario (where you want it to be `false`) don't pass in the prop --> make it not mandatory?

**Review by AmandaMisjuwar — APPROVED** [review]

**Review by annechung — APPROVED** [review]
lgtm!

**Pull request merged**